hack/bin/generate-local-nix-overrides.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+ROOT_DIR=$(cd -- "$SCRIPT_DIR/../../" &> /dev/null && pwd)
+cabalFiles=$(find "$ROOT_DIR" -name '*.cabal' \
+                 | grep -v dist-newstyle)
```

Comment:
Might have to use `-not -path` instead, since I think `dist-newstyle` doesn't appear in the final path component ("name").  Alternatively, you could type `\( -type d -name dist-newstyle -prune \) -o ...` to avoid walking those directories at all, if you are just worried about directories with that specific name and not something like "indist-newstyler.cabal".

Reply:
`\( -type d -name dist-newstyle -prune \) -o ...` doesn't seem to work and `-not -path` requires wildcards anyway. I think I will just leave it like this and hope nobody creates `indist-newstyler.cabal` or anything like that!